src/neptune_mlflow_exporter/impl/utils.py

```diff
@@ -0,0 +1,91 @@
+__all__ = [
+    "ArtifactUploadStrategy",
+    "FileUploadStrategy",
+    "DirectoryUploadStrategy",
+]
+
+import os
+import shutil
+from abc import (
+    ABC,
+    abstractmethod,
+)
+from pathlib import Path
+
+from mlflow.artifacts import download_artifacts
+from mlflow.entities import FileInfo
+
+try:
+    from neptune import Run
+except ImportError:
+    from neptune.new.metadata_containers import Run
+
+
+class ArtifactUploadStrategy(ABC):
+    BASE_NAMESPACE = "artifacts"
+
+    def __init__(
+        self,
+        *,
+        tracking_uri: str,
+        max_file_size: int,
+    ):
+
+        self._uri = tracking_uri
+        self._max_size = max_file_size
+
+    def _download_artifacts(self, run_id: str, path: str) -> str:
+        return download_artifacts(run_id=run_id, artifact_path=path, dst_path=".", tracking_uri=self._uri)
+
+    @abstractmethod
+    def upload_artifact(self, neptune_run: Run, info: FileInfo, run_id: str) -> None:
+        ...
+
+
+class FileUploadStrategy(ArtifactUploadStrategy):
+    def upload_artifact(self, neptune_run: Run, info: FileInfo, run_id: str) -> None:
+        if info.is_dir or info.file_size > self._max_size:
+            return
+
+        local_path = None
+        try:
+            local_path = self._download_artifacts(run_id=run_id, path=info.path)
+            local_path = Path(local_path).name
+            neptune_run[f"{self.BASE_NAMESPACE}/{local_path}"].upload(local_path)
+            neptune_run.wait()
+
+        finally:
+            if local_path and os.path.isfile(local_path):
+                os.remove(local_path)
+
+
+class DirectoryUploadStrategy(ArtifactUploadStrategy):
+    def get_dir_size(self, path: str):
+        total = 0
+        with os.scandir(path) as it:
```

Comment:
This could be simplified with `Path(...).glob('**/*')`. Check this comment: https://stackoverflow.com/a/1392549 ("Update 2019" section).